src/AzureAppConfigurationImpl.ts

```diff
@@ -176,34 +176,70 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
         return {
             requestTracingEnabled: this.#requestTracingEnabled,
             initialLoadCompleted: this.#isInitialLoadCompleted,
-            appConfigOptions: this.#options
+            appConfigOptions: this.#options,
+            isFailoverRequest: this.#isFailoverRequest
         };
     }
 
-    async #loadSelectedKeyValues(): Promise<ConfigurationSetting[]> {
-        const loadedSettings: ConfigurationSetting[] = [];
+    async #executeWithFailoverPolicy(funcToExecute) {
+        const clientWrappers = await this.#clientManager.getClients();
+        if (clientWrappers.length === 0) {
+            this.#clientManager.refreshClients();
+            throw new Error("No client is available to connect to the target App Configuration store.");
+        }
 
-        // validate selectors
-        const selectors = getValidKeyValueSelectors(this.#options?.selectors);
+        for (const clientWrapper of clientWrappers) {
+            let successful = false;
+            try {
+                const result = await funcToExecute(clientWrapper.client);
+                this.#isFailoverRequest = false;

```

Comment:
isFailoverRequest should be set to false before iterating all clients.

Let's say we call load() and then call refresh().

During load(), a failover happend, the first client failed and the second client failed. Now `#isFailoverRequest` is set to true.

Then, we call refresh(), we are going to send request to the second client, but this request will be tagged as failover request.

Reply:
`isFailoverRequest` is updated after each request. Once the current client succeeded, it will be set to `false` and next time the request will not be tagged failover.